discovery-infra/test_infra/utils/utils.py

```diff
@@ -789,6 +791,26 @@ def download_iso(image_url, image_path):
             out.write(chunk)
 
 
+def fetch_url(url, timeout=60, max_retries=10):
+    """
+    Returns the response content for the specified URL.
+    Raises an exception in case of any failure.
+    """
+    try:
+        retries = Retry(
+            read=max_retries,
+            status=max_retries,
+            backoff_factor=1,
+            status_forcelist=[500])
+        s = Session()
+        s.mount('http://', HTTPAdapter(max_retries=retries))
+        response = s.get(url, timeout=timeout)
+        response.raise_for_status()
+        return response.content
+    except Exception as err:
```

Comment:
suggestion:

`except requests.exceptions.RequestException as err:`

to be more specific with the error we want to catch